src/generator/AutoRest.Ruby.Azure/Model/CompositeTypeRba.cs

```diff
@@ -50,45 +50,62 @@ public override string GetBaseTypeName()
                 if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
                     this.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension))
                 {
-                    if(!nameRegEx.IsMatch(typeName) || !IsResourceModelsMatchStandardDefinition(this))
+                    if (!resourceOrSubResourceRegEx.IsMatch(typeName) || !IsResourceModelMatchingStandardDefinition(this))
                     {
                         typeName = "MsRestAzure::" + typeName;
                     }
                 }
 
                 return " < " + typeName;
             }
-            else if (nameRegEx.IsMatch(this.Name.ToString()))
+            else if (resourceOrSubResourceRegEx.IsMatch(this.Name))
             {
-                return " < " + "MsRestAzure::" + this.Name.ToString();
+                return " < " + "MsRestAzure::" + this.Name;
             }
 
             return string.Empty;
         }
 
-        public static bool IsResourceModelsMatchStandardDefinition(CompositeType model)
+        /// <summary>
+        /// Checks if the provided definition of models 'Resource'/'SubResource' matches the standard definition. 
```

Comment:
should the description look something like this: "Checks if the provided definition of model, matches the definition of 'Resource'/'SubResource' as defined in MsRestAzure" ?